### DISCUSSION TITLE
No cells.json

### DISCUSSION THREAD

**Discussion by mikepowerntie** [the1killer/sm_overview#11] [Q&A] [answered]
As the title already says, somehow i don't have a cells.json file in my scrap mechanic survival files.
Is there a way to fix this or am i just doomed?

**Comment by Itai-Iloni** [comment, marked as answer]
In 0.6.0, Scrap Mechanic overhauled their internal terrain database, meaning the1killer's map generation tool is obsolete in its current incarnation.

I was able to fix it following these steps, but they rely on the existence of legacyIds graciously kept by Axolot developers. Here was what I did:

1. Instead of copy pasting the code below into `Survival/Scripts/game/terrain/terrain_overworld.lua`, copy paste it into `Survival/Scripts/terrain/terrain_overworld.lua`
2. When copy pasting the code, you will need to change part of it:

```
                local cells = {}
		forEveryCell( function( cellX, cellY )
			local cell = {}
			cell["x"] = cellX
			cell["y"] = cellY;
			cell["tileid"] = GetLegacyID(GetCellTileUid( cellX, cellY )) -- modified by Arkanorian to work for update 0.6.0
			cell["flags"] = g_cellData.flags[cellY][cellX]
			cell["rotation"] = g_cellData.rotation[cellY][cellX]
			-- cell["elevation"] = g_cellData.elevation[cellY][cellX]
			-- cell["clifflevel"] = g_cellData.cliffLevel[cellY][cellX]
			-- cell["offx"] = g_cellData.tileOffsetX[cellY][cellX]
			-- cell["offy"] = g_cellData.tileOffsetY[cellY][cellX]
			-- cell["celldebug"] = g_cellData.cellDebug[cellY][cellX]
			-- cell["cornerdebug"] = g_cellData.cornerDebug[cellY][cellX]
			-- cell["variation"] = sm.noise.intNoise2d( cellX, cellY, g_cellData.seed + 2854 )

			cells[#cells+1] = cell
		end )
		if #cells > 0 then
			cells[1]["bounds"] = g_cellData.bounds
			cells[1]["seed"] = g_cellData.seed
			-- print("Writing Cell Json");
			sm.json.save( cells, "$SURVIVAL_DATA/".."cells.json" )
			cells = nil;
			-- print("Wrote Cell Json");
		end
```
The modification is on the line starting with `cell["tileid"]`

3. You will need to modify `Survival/Scripts/terrain/overworld/tile_database.lua`:

- Add `local legacyIds = {}` under the line `local f_idToTilePath = {}`
- Add the following after the end of`function GetPoiType( uid )`:
```
-- Added by Arkanorian for use in 0.6.0
function GetLegacyID( uid )
	if not legacyIds[tostring( uid )] then
		return nil
	end
	return legacyIds[tostring( uid )]
end
```
- Add `legacyIds[tostring( uid )] = legacyId` under the line `AddLegacyUpgrade( legacyId, uid )`

After making these changes, cells.json should appear and have the information necessary for this map generator to work. Follow the steps in the README.MD and the map will come out properly.

I created and updated version of the original [Readme.md](https://github.com/the1killer/sm_overview/files/8847338/Readme.md) for all of the steps together. The two new files are attached below, but I can only upload them as .txt, so change them to .lua before using them.
[terrain_overworld.txt](https://github.com/the1killer/sm_overview/files/8847492/terrain_overworld.txt)
[tile_database.txt](https://github.com/the1killer/sm_overview/files/8847506/tile_database.txt)

**Reply by the1killer** [reply]
@Itai-Iloni can you fork the repo and do a pull request?
